leo/core/leoAtFile.py

```diff
@@ -2991,24 +3010,25 @@ def replaceFile(self,
             # No original file to change. Return value tested by a unit test.
             return False  # No change to original file.
 
-        # Compare the old and new contents.
-        old_contents = g.readFileIntoUnicodeString(fileName,
-            encoding=at.encoding, silent=True)
-        if not old_contents:
-            old_contents = ''
-        unchanged = (
-            contents == old_contents
-            or (not at.explicitLineEnding and at.compareIgnoringLineEndings(old_contents, contents))
-            or ignoreBlankLines and at.compareIgnoringBlankLines(old_contents, contents))
-        if unchanged:
-            at.unchangedFiles += 1
-            if not g.unitTesting and c.config.getBool(
-                'report-unchanged-files', default=True):
-                g.es(f"{timestamp}unchanged: {sfn}")  # pragma: no cover
-            # Check unchanged files.
-            at.checkPythonCode(contents, fileName, root)
-            return False  # No change to original file.
-        #
+        if not root.isAtCleanNode():  # #4394: Always write @clean nodes!
+            # Compare the old and new contents.
+            old_contents = g.readFileIntoUnicodeString(fileName,
```

Comment:
oh oh: **old_contents**  has to be set BEFORE this` if not root.isAtCleanNode(): ` line above because it is also used after that 'if' below, at line 3035. (would be completely unset otherwise)